code/harness/Tester.cpp

```diff
@@ -194,6 +197,53 @@ int Tester::umount_device() {
   return SUCCESS;
 }
 
+int Tester::mount_snapshot() {
```

Comment:
Is there a way to modify some of the existing mount/umount code to take more arguments so we don't have to define yet another set of functions for it that also call the system calls? Or can we modify already existing functions to call these with the right arguments? I would prefer to start trimming down the places we have system calls like this sprinkled throughout the code if we can